File: Study Plan/Algorithm I/Flood_Fill.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, newColor: int) -> List[List[int]]:
        if image == None or image[sr][sc] == newColor:
            return image
        self.fill(image, sr, sc, image[sr][sc], newColor)
        return image
        
        
    def fill(self, image, row, column, initial, new_color):
        if row < 0 or row >= len(image) or column < 0 or column >= len(image[0])  or image[row][column] != initial:
            return
        
        image[row][column] = new_color
        self.fill(image, row + 1, column, initial, new_color)
        self.fill(image, row - 1,column , initial, new_color)
        self.fill(image, row, column + 1, initial, new_color)
        self.fill(image, row, column - 1, initial, new_color)
        

File: Study Plan/Algorithm I/Flood_Fill.cpp
class Solution {
public:
    void dfs(vector<vector<int>> &grid,int x,int y, int val , int newColor){
        if(x < 0 || x >= grid.size() || y < 0 || y >= grid[0].size() || grid[x][y] != val)
            return;
        
        grid[x][y] = newColor;
        
        dfs(grid, x + 1, y, val, newColor);
        dfs(grid, x, y + 1, val, newColor);
        dfs(grid, x - 1, y, val, newColor);
        dfs(grid, x, y - 1, val, newColor);   
    }
    
    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int newColor) {
        if (image[sr][sc] == newColor) 
            return image;
        dfs(image, sr, sc, image[sr][sc], newColor);
        return image;
    }
};